pandas/io/common.py

```diff
@@ -583,66 +538,73 @@ def get_handle(
         Specifies how encoding and decoding errors are to be handled.
         See the errors argument for :func:`open` for a full list
         of options.
+    storage_options: StorageOptions = None
+        Passed to _get_filepath_or_buffer
 
     .. versionchanged:: 1.2.0
 
     Returns the dataclass IOHandles
     """
-    need_text_wrapping: Tuple[Type["IOBase"], ...]
-    try:
-        from s3fs import S3File
-
-        need_text_wrapping = (BufferedIOBase, RawIOBase, S3File)
-    except ImportError:
-        need_text_wrapping = (BufferedIOBase, RawIOBase)
-    # fsspec is an optional dependency. If it is available, add its file-object
-    # class to the list of classes that need text wrapping. If fsspec is too old and is
-    # needed, get_filepath_or_buffer would already have thrown an exception.
-    try:
-        from fsspec.spec import AbstractFileSystem
-
-        need_text_wrapping = (*need_text_wrapping, AbstractFileSystem)
```

Comment:
`fssepc` objects and `S3File` all have the attribute `mode`. This allows users to create `fsspec` objects in text mode and use them with pandas.